src/dispatch.c

```diff
@@ -315,15 +368,40 @@ void nats_dispatchMessages(natsDispatcher *d)
             natsConn_removeSubscription(nc, sub);
             continue;
         }
+        else if ((fetchStatus != NATS_OK) && !lastMessageInFetch)
+        {
+            // If last message in fetch, will call the callback after the
+            // message is delievered.
 
-        // --- Real messages (user or flow control/status) ---
+            // FIXME: do we care to send an async error, if serious?
```

Comment:
Basically, all errors currently result in termination of the Fetch/sub, and fetchCompleteCB is called with the error.

I would like to add recovery (at least from Re-connect, and missed HB), but as a separate PR. I removed the FIXMEs because a (idiomatic?) mechanism to get notified exists.